.circleci/config.yml

```diff
@@ -0,0 +1,65 @@
+####################
+# CircleCI configuration reference:
+#   https://circleci.com/docs/2.0/configuration-reference
+####################
+
+version: 2
+
+#####################################################
+# Jobs: see https://circleci.com/docs/2.0/jobs-steps/
+#####################################################
+
+jobs:
+  build:
+    docker: # run the steps with Docker
+      - image: mozilla/cidockerbases:docker-latest
+    working_directory: /dockerflow
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build Docker image
+          command: docker build -t app:build .
+
+      # save the built docker container into CircleCI's cache. This is
+      # required since Workflows do not have the same remote docker instance.
```

Comment:
I think you'd be better served by the [persist_to_workspace](https://circleci.com/docs/2.0/configuration-reference/#persist_to_workspace) and attach_workspace steps. I'm not sure CircleCI provides sufficient guarantees around cache behavior to allow a cache in one job to be available reliably to a subsequent job in the same workflow. It would also remove the need to define a key consistently between the two jobs.

Reply:
This would actually be much simpler with https://circleci.com/docs/2.0/docker-layer-caching/ but I don't know if we have it enabled on our account. With docker layer caching, we could see significantly faster build times in the normal case, and we could skip explicitly caching from one step to the next in a workflow.

Reply:
It's now not clear to me why these are separate jobs at all. Could the workflow be just a single `test` job that handles building the image and then running the tests all in one?

Reply:
Agreed, without a deploy this is unnecessary. Switching to a single step.